frontend/projects/upgrade/src/app/features/dashboard/segments/modals/upsert-segment-modal/upsert-segment-modal.component.ts

```diff
@@ -268,6 +270,25 @@ export class UpsertSegmentModalComponent {
     this.segmentService.modifySegment(segmentRequest);
   }
 
+  createDuplicateRequest({ name, description, tags }: SegmentFormData, sourceSegment: Segment): void {
+    const userIds = sourceSegment.individualForSegment.map((user) => user.userId);
+    const groups = sourceSegment.groupForSegment;
+    const subSegmentIds = sourceSegment.subSegments.map((subSegment) => subSegment.id);
```

Comment:
We need the actual segments here, not just the IDs. We should be re-creating the private segments for each duplicate, not sharing them. This also means that we'll probably need a different endpoint for this. It could be very similar.